### ISSUE TITLE
[BUG] clang trunk arm64 headers see inconsistent state

### ISSUE THREAD

**Issue by mmdriley** [compiler-explorer/compiler-explorer#2258]
In particular, `uintptr_t` is not pointer-sized.

repro: https://godbolt.org/z/3MK35a

**Comment by mattgodbolt** [comment]
Notes (mostly to myself): `sizeof(void*)` is 8, but `sizeof(uintptr_t)` is 4. The command flags we use are: 

```
-g -o /tmp/compiler-explorer-compiler202095-893-1j5hn6n.qdli/output.s -S --gcc-toolchain=/opt/compiler-explorer/gcc-9.2.0 -target aarch64-none-linux-android -mtune=cortex-a53 -march=armv8.5-a+crypto+rcpc+sha3+sm4+profile+rng+memtag+sve2+sve2-bitperm+sve2-sm4+sve2-aes+sve2-sha3+tme -fcolor-diagnostics -fno-crash-diagnostics -std=c++17 /tmp/compiler-explorer-compiler202095-893-1j5hn6n.qdli/example.cpp
```

Looks like we're getting system headers still: https://godbolt.org/z/4e4q3T

```
# 1 "./example.cpp"
# 1 "<built-in>" 1
# 1 "<built-in>" 3
# 436 "<built-in>" 3
# 1 "<command line>" 1
# 1 "<built-in>" 2
# 1 "./example.cpp" 2
# 1 "/opt/compiler-explorer/clang-trunk-20201005/lib/clang/12.0.0/include/stdint.h" 1 3
# 52 "/opt/compiler-explorer/clang-trunk-20201005/lib/clang/12.0.0/include/stdint.h" 3
# 1 "/usr/include/stdint.h" 1 3 4
# 26 "/usr/include/stdint.h" 3 4
# 1 "/usr/include/bits/libc-header-start.h" 1 3 4
# 33 "/usr/include/bits/libc-header-start.h" 3 4
# 1 "/usr/include/features.h" 1 3 4
# 439 "/usr/include/features.h" 3 4
# 1 "/usr/include/stdc-predef.h" 1 3 4
# 440 "/usr/include/features.h" 2 3 4
# 461 "/usr/include/features.h" 3 4
# 1 "/usr/include/sys/cdefs.h" 1 3 4
# 452 "/usr/include/sys/cdefs.h" 3 4
# 1 "/usr/include/bits/wordsize.h" 1 3 4
# 453 "/usr/include/sys/cdefs.h" 2 3 4
# 1 "/usr/include/bits/long-double.h" 1 3 4
# 454 "/usr/include/sys/cdefs.h" 2 3 4
# 462 "/usr/include/features.h" 2 3 4
# 485 "/usr/include/features.h" 3 4
# 1 "/usr/include/gnu/stubs.h" 1 3 4

# 1 "/usr/include/gnu/stubs-32.h" 1 3 4
# 8 "/usr/include/gnu/stubs.h" 2 3 4
# 486 "/usr/include/features.h" 2 3 4
# 34 "/usr/include/bits/libc-header-start.h" 2 3 4
# 27 "/usr/include/stdint.h" 2 3 4
# 1 "/usr/include/bits/types.h" 1 3 4
# 27 "/usr/include/bits/types.h" 3 4
# 1 "/usr/include/bits/wordsize.h" 1 3 4
# 28 "/usr/include/bits/types.h" 2 3 4
# 1 "/usr/include/bits/timesize.h" 1 3 4
# 29 "/usr/include/bits/types.h" 2 3 4
...
```

**Comment by partouf** [comment]
Basically related to https://github.com/compiler-explorer/compiler-explorer/issues/2018

Even though this is Clang trunk, we have no bits headers for arm, they need to come from either Android NDK or from the Arm GCC's.

**Comment by mattgodbolt** [comment]
We have a number of cross-compilation errors. We don't actually have the OS headers for non-x86 headers available, so this is purely clang generating ARM code, not an ARM development system. See also #915, #2131, #2018, and no doubt many more :|

We should do better at cross-compilation; sorry. We don't have a good story for cross-compilation; it's quite enough for us to keep up with x86 linux compilers.

**Comment by willlovett-arm** [comment]
Hi folks.  I think I have a solution.  We need to fix the target, and need to make use of the arm64 gnu cross compiler you already have, rather than the x86 gnu compiler.

Here's an example of the original issue, with a triple, gcc-toolchain and sysroot that work: https://godbolt.org/z/bMoja8

Thanks to @mattgodbolt for showing me how to use your infra repo, from which I could work out the exact paths.

I will hope to make time to write a fixed config file patch either today or next week!

**Comment by willlovett-arm** [comment]
I think #2268 should fix this

**Comment by partouf** [comment]
This is now fixed by using #2268 - let us know if you have any issues!
